android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinLoginScreen.kt

```diff
@@ -0,0 +1,272 @@
+import androidx.compose.material.icons.outlined.MoreVert
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.smartregister.fhircore.engine.R
+import org.smartregister.fhircore.engine.ui.components.PinView
+import org.smartregister.fhircore.engine.ui.login.APP_LOGO_TAG
+import org.smartregister.fhircore.engine.ui.theme.LoginButtonColor
+import org.smartregister.fhircore.engine.ui.theme.LoginDarkColor
+import org.smartregister.fhircore.engine.util.annotation.ExcludeFromJacocoGeneratedReport
+
+const val PIN_TOOLBAR_MENU = "toolbarMenuTag"
+const val PIN_TOOLBAR_MENU_BUTTON = "toolbarMenuButtonTag"
+const val PIN_TOOLBAR_TITLE = "toolbarTitle"
+const val PIN_TOOLBAR_MENU_ICON = "toolbarIcon"
+const val PIN_TOOLBAR_MENU_LOGIN = "toolbarMenuLogin"
+const val PIN_FORGOT_PIN = "forgotPin"
+const val PIN_FORGOT_DIALOG = "forgotPinDialog"
+const val PIN_TOOLBAR_MENU_SETTINGS = "toolbarMenuSettings"
+const val PIN_SET_PIN_CONFIRM_BUTTON = "setPinConfirmButton"
+
+@Composable
+fun PinLoginScreen(viewModel: PinViewModel) {
+
+  val showError by viewModel.showError.observeAsState(initial = false)
+
+  PinLoginPage(
+    onPinChanged = viewModel::onPinChanged,
+    showError = showError,
+    enterUserPinMessage = viewModel.enterUserLoginMessage,
+    onMenuLoginClicked = { viewModel.onMenuLoginClicked() },
+    forgotPin = viewModel::forgotPin,
+  )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun PinLoginPage(
+  modifier: Modifier = Modifier,
+  onPinChanged: (String) -> Unit,
+  showError: Boolean = false,
+  onMenuLoginClicked: () -> Unit,
+  enterUserPinMessage: String = "",
+  forgotPin: () -> Unit
+) {
+
+  var showMenu by remember { mutableStateOf(false) }
+  var showForgotPinDialog by remember { mutableStateOf(false) }
+
+  Surface(color = colorResource(id = R.color.white_slightly_opaque)) {
+    TopAppBar(
+      title = { Text(text = "", Modifier.testTag(PIN_TOOLBAR_TITLE)) },
+      navigationIcon = {
+        IconButton(onClick = {}) {
+          Icon(
+            Icons.Filled.ArrowBack,
+            contentDescription = "Back arrow",
```

Comment:
this text is for view reference only, it will not be reflecting in application and doesn't required MLS.
so no need to add extra string in string.xml